packages/playground/metro.config.js

```diff
@@ -7,11 +7,106 @@
+function isRelativeImport(filePath) {
+  return /^[.][.]?(?:[/]|$)/.test(filePath);
+}
+
+// Example: vnextResolve('C:/Repos/react-native-windows/vnext/', './Libraries/Text/Text');
+// Returns a full path to the resolved location which would be in vnext/src if the file exists there
+// or just in vnext otherwise
+function vnextResolve(originDir, moduleName) {
+  const originDirSrc = originDir.replace(
+    rnwPathNormalized,
+    rnwSrcPathNormalized,
+  );
+
+  // redirect the resolution to src if an appropriate file exists there
+  const extensions = [
+    '.windows.tsx',
+    '.windows.ts',
+    '.windows.jsx',
+    '.windows.js',
+    '.tsx',
+    '.ts',
+    '.jsx',
+    '.js',
+  ];
+
+  // For each potential extension we need to check for the file in either vnext/src, or in vnext
+  for (let extension of extensions) {
+    const potentialSrcModuleName =
+      path.resolve(originDirSrc, moduleName) + extension;
+    if (fs.existsSync(potentialSrcModuleName)) {
+      return potentialSrcModuleName;
+    }
+
+    const potentialModuleName = path.resolve(originDir, moduleName) + extension;
+    if (fs.existsSync(potentialModuleName)) {
+      return potentialModuleName;
+    }
+  }
+}
+
+// This teaches metro how we merge vnext/src and files from core to vnext...
+// Basically any resolves within vnext we look to see if the file exists in vnext/src,
+// and use that one if it exists.  That way fast refresh will work on files within vnext/src
+function vnextDevResolveRequest(
+  context,
+  _realModuleName /* string */,
+  platform /* string */,
+  moduleName /* string */,
+) {
+  let backupResolveRequest = context.resolveRequest;
+  delete context.resolveRequest;
+
+  try {
+    let modifiedModuleName = moduleName;
+    if (platform === 'windows') {
+      if (moduleName === 'react-native') {
+        // redirect react-native -> vnext/src/index.windows.js
+        modifiedModuleName = path.resolve(rnwPath, 'src/index.windows.js');
+      } else if (moduleName.startsWith('react-native/')) {
+        // redirect react-native/foo -> vnext/src/foo or vnext/foo
+        modifiedModuleName = vnextResolve(
+          rnwPath,
+          (moduleName = `./${moduleName.slice('react-native/'.length)}`),
+        );
+      } else if (
+        context.originModulePath.startsWith(rnwPath) &&
+        isRelativeImport(moduleName)
+      ) {
+        // We need to handle relative paths from within react-native-windows
+        let originPathWithoutSrc = context.originModulePath;
+        if (originPathWithoutSrc.startsWith(rnwSrcPathNormalized))
+          originPathWithoutSrc = `${rnwPath}/${originPathWithoutSrc.slice(
+            rnwSrcPathNormalized.length,
+          )}`;
+
+        const resolvedLocation = vnextResolve(
+          path.dirname(originPathWithoutSrc),
+          moduleName,
+        );
+        if (resolvedLocation) modifiedModuleName = resolvedLocation;
+      }
+    }
+    let result = resolve(context, modifiedModuleName, platform);
```

Comment:
since this is the fallback for other modules, should `context.resolveRequest` be restored before going down the resolve here?

Reply:
No we delete that to ensure that we go down a specific code path within metro-resolver.resolve.